error: invalid conversion from 'int*' to 'mat_uint32_t*'

Hi all.

`openmeeg-2.4.2` is not compiling against the new `matio-1.5.18`:

```
[  9%] Building CXX object OpenMEEGMaths/CMakeFiles/OpenMEEGMaths.dir/src/MatlabIO.C.o
cd /builddir/build/BUILD/openmeeg-2.4.2/build/OpenMEEGMaths && /usr/bin/g++ -DHAVE_BLAS -DHAVE_LAPACK -DOpenMEEGMaths_EXPORTS -DUSE_LAPACK -I/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include -I/usr/include/flexiblas -I/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/MATIO::MATIO -I/builddir/build/BUILD/openmeeg-2.4.2/build/OpenMEEGMaths -I/builddir/build/BUILD/openmeeg-2.4.2/build -I/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/OpenMEEGMaths/src -O2  -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -DNDEBUG -fPIC -std=gnu++11 -o CMakeFiles/OpenMEEGMaths.dir/src/MatlabIO.C.o -c /builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/src/MatlabIO.C
In file included from /builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/src/MatlabIO.C:40:
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H: In member function 'void OpenMEEG::maths::MatlabIO::write_sparse(mat_t*, const OpenMEEG::LinOp&) const':
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:42: warning: narrowing conversion of '(int)sz' from 'int' to 'mat_uint32_t' {aka 'unsigned int'} [-Wnarrowing]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                          ^~
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:46: error: invalid conversion from 'int*' to 'mat_uint32_t*' {aka 'unsigned int*'} [-fpermissive]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                              ^~
      |                                              |
      |                                              int*
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:50: warning: narrowing conversion of '(int)sz' from 'int' to 'mat_uint32_t' {aka 'unsigned int'} [-Wnarrowing]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                                  ^~
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:54: error: invalid conversion from 'int*' to 'mat_uint32_t*' {aka 'unsigned int*'} [-fpermissive]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                                      ^~
      |                                                      |
      |                                                      int*
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:58: warning: narrowing conversion of '(int)num' from 'int' to 'mat_uint32_t' {aka 'unsigned int'} [-Wnarrowing]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                                          ^~~
/builddir/build/BUILD/openmeeg-2.4.2/OpenMEEGMaths/include/MatlabIO.H:384:63: warning: narrowing conversion of '(int)sz' from 'int' to 'mat_uint32_t' {aka 'unsigned int'} [-Wnarrowing]
  384 |                 mat_sparse_t  sparse = { sz, ir, sz, jc, num, sz, t};
      |                                                               ^~
make[2]: *** [OpenMEEGMaths/CMakeFiles/OpenMEEGMaths.dir/build.make:176: OpenMEEGMaths/CMakeFiles/OpenMEEGMaths.dir/src/MatlabIO.C.o] Error 1
make[2]: Leaving directory '/builddir/build/BUILD/openmeeg-2.4.2/build'
make[1]: *** [CMakeFiles/Makefile2:380: OpenMEEGMaths/CMakeFiles/OpenMEEGMaths.dir/all] Error 2
make[1]: Leaving directory '/builddir/build/BUILD/openmeeg-2.4.2/build'
make: *** [Makefile:163: all] Error 2
```

Fedora 34 (devel branch)
GCC-10.2.1

--- a/OpenMEEGMaths/include/MatlabIO.H
+++ b/OpenMEEGMaths/include/MatlabIO.H
@@ -352,12 +352,19 @@ namespace OpenMEEG {
                     tank_inverted[std::pair<size_t,size_t>(j,i)] = val;
                 }
 
-                const int sz = linop.size();
-                const int num = linop.ncol()+1;
 
-                double* t  = new double[sz];
-                int*    ir = new int[sz];
-                int*    jc = new int[num];
+                #if MATIO_VERSION>=1518
+                    typedef mat_uint32_t matio_int_type;
+                #else
+                    typedef int matio_int_type;
+                #endif
+
+                const matio_int_type sz  = linop.size();
+                const matio_int_type num = linop.ncol()+1;
+                matio_int_type*      ir  = new matio_int_type[sz];
+                matio_int_type*      jc  = new matio_int_type[num];
+
+                double* t = new double[sz];
 
                 size_t cnt = 0;
                 size_t current_col = (size_t)-1;

